eclipse-theia/theia
# WIP: Implement order for info-marker-nodes

#### What it does
Fixes #8983

Previously info-marker-nodes were ordered by the time they have been reported to the problem-tree-model. This led to an unexpected order of the nodes in the problem view. (See #8983)
In this commit info-marker-nodes will be inserted to the tree-model based on the following rules:
-nodes will be sorted based on the most severe marker they contain
-if nodes contain equally severe markers, they are sorted by the URI

#### How to test
Download test workspace from here:
https://github.com/eclipse-theia/theia/files/5859357/problems-view.zip

You should be able to: 

- Open and close all files. The nodes concerning the specific file should be added in the defined order to the previous nodes.

- Change settings without the order of nodes being changed.

- Add or remove warning or errors to a file. The order in the problem-view should be adjusted.

- Rename files. The order in the problem-view should be adjusted.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
